patches/server/1029-Reduce-work-done-in-CraftMapCanvas.drawImage-by-limi.patch

```diff
@@ -0,0 +1,103 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Barnaby <22575741+barnabwhy@users.noreply.github.com>
+Date: Sat, 29 Jun 2024 12:06:51 +0100
+Subject: [PATCH] Reduce work done in CraftMapCanvas.drawImage by limiting size
+ of image and using System.arraycopy instead of for loops and use bitwise
+ operations to do bounds checks.
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/map/CraftMapCanvas.java b/src/main/java/org/bukkit/craftbukkit/map/CraftMapCanvas.java
+index ff59f759669620795ef355c988b664bdcda39f52..59f61beb0f179d543fa6adab0989e676a06ab802 100644
+--- a/src/main/java/org/bukkit/craftbukkit/map/CraftMapCanvas.java
++++ b/src/main/java/org/bukkit/craftbukkit/map/CraftMapCanvas.java
+@@ -1,8 +1,10 @@
+ package org.bukkit.craftbukkit.map;
+ 
+ import com.google.common.base.Preconditions;
+ import java.awt.Color;
++import java.awt.Graphics2D;
```

Comment:
Usually imports should generally be avoided in non paper classes to reduce diff ;)